src/ui/components/tenants/tenantDetail/thirdPartyPage/ThirdPartyPage.tsx

```diff
@@ -0,0 +1,436 @@
+ * under the License.
+ */
+import { Dispatch, SetStateAction, useEffect, useState } from "react";
+import { useGetThirdPartyProviderInfo } from "../../../../../api/tenants";
+import { ProviderConfigResponse, TenantDashboardView } from "../../../../../api/tenants/types";
+import { getImageUrl, isValidHttpUrl } from "../../../../../utils";
+import Button from "../../../button";
+import { Loader } from "../../../loader/Loader";
+import { useTenantDetailContext } from "../TenantDetailContext";
+import { TenantDetailHeader } from "../TenantDetailHeader";
+import { PanelHeader, PanelHeaderTitleWithTooltip, PanelRoot } from "../tenantDetailPanel/TenantDetailPanel";
+import { ProviderInfoForm } from "../thirdPartyProviderConfig/ProviderInfoForm";
+import { ThirdPartyProviderInput } from "../thirdPartyProviderInput/ThirdPartyProviderInput";
+import "./thirdPartyPage.scss";
+
+export const ThirdPartyPage = ({
+	viewObj,
+	setViewObj,
+}: {
+	viewObj: TenantDashboardView;
+	setViewObj: Dispatch<SetStateAction<TenantDashboardView>>;
+}) => {
+	const handleProviderInfoBack = () => {
+		setViewObj({ view: "tenant-detail" });
+	};
+	return (
+		<div className="third-party-section">
+			<button
+				className="button flat"
+				onClick={() => handleProviderInfoBack()}>
+				<img
+					src={getImageUrl("left-arrow-dark.svg")}
+					alt="Go back"
+				/>
+				<span>Back to tenant info</span>
+			</button>
+			<div className="third-party-section__cards">
+				<TenantDetailHeader onlyShowTenantId />
+				{viewObj.view === "add-or-edit-third-party-provider" && (
+					<ProviderInfo
+						providerId={viewObj.thirdPartyId}
+						isAddingNewProvider={viewObj.isAddingNewProvider}
+						handleGoBack={handleProviderInfoBack}
+					/>
+				)}
+			</div>
+		</div>
+	);
+};
+
+const ProviderInfo = ({
+	providerId,
+	isAddingNewProvider,
+	handleGoBack,
+}: {
+	providerId?: string;
+	isAddingNewProvider: boolean;
+	handleGoBack: (shouldGoBackToDetailPage?: boolean) => void;
+}) => {
+	const { tenantInfo } = useTenantDetailContext();
+	const [isProviderInfoLoading, setIsProviderInfoLoading] = useState(true);
+	const [hasFilledCustomFieldsForProvider, setHasFilledCustomFieldsForProvider] = useState(
+		!PROVIDERS_WITH_ADDITIONAL_CONFIG.includes(providerId ?? "")
+	);
+	const getThirdPartyProviderInfo = useGetThirdPartyProviderInfo();
+	const [providerConfigResponse, setProviderConfigResponse] = useState<ProviderConfigResponse | undefined>(undefined);
+	const providerHasCustomFields =
+		typeof providerId === "string" && PROVIDERS_WITH_ADDITIONAL_CONFIG.includes(providerId);
+
+	const fetchProviderInfo = async (id: string, additionalConfig?: Record<string, string>) => {
+		setHasFilledCustomFieldsForProvider(true);
+		try {
+			setIsProviderInfoLoading(true);
+			const response = await getThirdPartyProviderInfo(tenantInfo.tenantId, id, additionalConfig);
+			if (response.status === "OK") {
+				setProviderConfigResponse(response.providerConfig);
+			}
+		} catch (error) {
```

Comment:
- we need to handle okta invalid domains by showing an error message to the user, and maybe a something went wrong in the UI.
- we also need to test what happens if okta is defined in a static config with an invalid domain, and we click on add new okta provider
- we also need to test what happens if okta is defined in a static config with a valid domain, and we click on add new okta provider and the user enters a different okta domain.

Reply:
@sattvikc will do / test above